web/js/general.js

```diff
@@ -0,0 +1,28 @@
+$(document).ready(function () {
+    $('html').removeClass('no-js');
+
+    // To prevent duplicates, use the first locale options as a reference for building buttons
+    var localeOptions = $('.locale-switch:first option');
+
+    for (var i = 0; i < localeOptions.length; i++) {
+        var localeDataAttribute = 'data-locale="'+ localeOptions[i].value +'"';
+
+        var localeOptionClassAttribute = '';
+        if (localeOptions[i].selected) {
+            localeOptionClassAttribute = 'class="active"';
+        }
+
+        var localeOptionButton =
+            '<button ' + localeOptionClassAttribute + ' ' + localeDataAttribute + '>'
+            + localeOptions[i].text
+            + '</button>';
+        $('.locale-switch').append(localeOptionButton);
+    }
+
+    $(document).on('click', '.locale-switch button[data-locale]' , function(e) {
+        var localeChoice = $(e.target).attr('data-locale');
+        $('.locale-switch:first option[value='+ localeChoice +']').attr('selected', true);
```

Comment:
0. A potential bug if `data-locale` should ever contain something like an apostrophe. Unlikely, but use jQuery's `filter()` function.
0. Also, the proper way to address boolean attributes, such as `selected` and `checked`, is using properties. This becomes apparent when attempting to deselect (or uncheck) elements, and you have to actually *remove* the attribute to deselect the element. See jQuery's `prop` function.